src/webui/api/synccontroller.cpp

```diff
@@ -550,7 +550,10 @@ void SyncController::torrentPeersAction()
             peer[KEY_PEER_COUNTRY] = Net::GeoIPManager::CountryName(pi.country());
         }
 
-        peers[pi.address().ip.toString() + ':' + QString::number(pi.address().port)] = peer;
+        if (!pi.address().ip.toString().contains('.'))  // IPv6
```

Comment:
It's possible for some representations of IPv6 addresses to contain periods. For example, `::ffff:192.0.2.128` is a valid IPv4-mapped IPv6 address. I think a more reliable check would be to use the presence of a colon `:` to indicate that the address is IPv6. This would also be consistent with the logic used in other places in qBittorrent.

Reply:
Find that the similar logic has been used in `peerlistwidgets.cpp`. It seems that it was introduced by @ngosang. Considering the IPv4-mapped IPv6 address, we may also need to modify these codes.
https://github.com/qbittorrent/qBittorrent/blob/master/src/gui/properties/peerlistwidget.cpp#L330-L333